### PR TITLE
Blaze: Ad destination parameters

### PR DESCRIPTION
**Pull request by 0nko**
Implements #10737, a subtask of #10665.

[Screen_recording_20240209_213359.webm](https://github.com/woocommerce/woocommerce-android/assets/1522856/11b50071-b42b-4c1a-b92f-d21e80c863cb)

**To test:**
1. Apply the patch below to see any parameters
2. Create a new Blaze campaign
3. Tap on the Ad destination property
4. Notice the base URL and parameters are printed out
5. Tap on the URL parameters property
6. Notice the 2 parameters are displayed in a list
7. Notice a full URL is shown below the list
8. Delete one parameter
9. Notice the full URL and remaining character count is updated
10. Tap on back
11. Notice the parameters property got updated

**Patch:**
```kotlin
Index: WooCommerce/src/main/kotlin/com/woocommerce/android/ui/blaze/creation/preview/BlazeCampaignCreationPreviewFragment.kt
IDEA additional info:
Subsystem: com.intellij.openapi.diff.impl.patch.CharsetEP
<+>UTF-8
===================================================================
diff --git a/WooCommerce/src/main/kotlin/com/woocommerce/android/ui/blaze/creation/preview/BlazeCampaignCreationPreviewFragment.kt b/WooCommerce/src/main/kotlin/com/woocommerce/android/ui/blaze/creation/preview/BlazeCampaignCreationPreviewFragment.kt
--- a/WooCommerce/src/main/kotlin/com/woocommerce/android/ui/blaze/creation/preview/BlazeCampaignCreationPreviewFragment.kt	(revision Staged)
+++ b/WooCommerce/src/main/kotlin/com/woocommerce/android/ui/blaze/creation/preview/BlazeCampaignCreationPreviewFragment.kt	(date 1707510168898)
@@ -77,7 +77,7 @@
                 is NavigateToAdDestinationScreen -> findNavController().navigateSafely(
                     BlazeCampaignCreationPreviewFragmentDirections
                         .actionBlazeCampaignCreationPreviewFragmentToBlazeCampaignCreationAdDestinationFragment(
-                            event.targetUrl,
+                            event.targetUrl + "?test=true&yes=no",
                             event.productId
                         )
                 )
```

### CODE REVIEW

**Comment by dangermattic** [comment]
<table>
  <thead>
    <tr>
      <th width="50"></th>
      <th width="100%" data-danger-table="true" data-kind="Warning">
          3 Warnings
      </th>
     </tr>
  </thead>
  <tbody>
    <tr>
      <td>:warning:</td>
      <td data-sticky="false">This PR is larger than 300 lines of changes. Please consider splitting it into smaller PRs for easier and faster reviews.</td>
    </tr>
    <tr>
      <td>:warning:</td>
      <td data-sticky="false">Class <code>BlazeCampaignCreationAdDestinationParametersViewModel</code> is missing tests, but <code>unit-tests-exemption</code> label was set to ignore this.</td>
    </tr>
    <tr>
      <td>:warning:</td>
      <td data-sticky="false">Class <code>ViewState</code> is missing tests, but <code>unit-tests-exemption</code> label was set to ignore this.</td>
    </tr>
  </tbody>
</table>

<p align="right" data-meta="generated_by_pr-check">
  Generated by :no_entry_sign: <a href="https://danger.systems/">Danger</a>
</p>

File: WooCommerce/src/main/kotlin/com/woocommerce/android/ui/blaze/creation/destination/BlazeCampaignCreationAdDestinationParametersScreen.kt [hunk 1]
```diff
@@ -0,0 +1,221 @@
...
+                    }
+                }
+            }
+
+            item(key = "footer") {
+                Column(
+                    modifier = Modifier
+                        .animateItemPlacement()
+                        .fillMaxWidth()
+                ) {
```
Comment (JorgeMucientes) [inline WooCommerce/src/main/kotlin/com/woocommerce/android/ui/blaze/creation/destination/BlazeCampaignCreationAdDestinationParametersScreen.kt:151]: Out of curiosity why did you add the footer inside the lazy column? Why not just below it?

File: WooCommerce/src/main/kotlin/com/woocommerce/android/ui/blaze/creation/destination/BlazeCampaignCreationAdDestinationParametersViewModel.kt [hunk 2]
```diff
@@ -0,0 +1,65 @@
...
+import kotlinx.coroutines.flow.MutableStateFlow
+import kotlinx.coroutines.flow.update
+import javax.inject.Inject
+
+@HiltViewModel
+class BlazeCampaignCreationAdDestinationParametersViewModel @Inject constructor(
+    savedStateHandle: SavedStateHandle
+) : ScopedViewModel(savedStateHandle) {
+    companion object {
+        // The maximum number of characters allowed in a URL by Chrome
```
Comment (JorgeMucientes) [inline WooCommerce/src/main/kotlin/com/woocommerce/android/ui/blaze/creation/destination/BlazeCampaignCreationAdDestinationParametersViewModel.kt:21]: Are we following the Chrome restrictions in the end or the Calypso web restrictions? Just double checking after seeing this convo: p1707498869846739-slack-C03L1NF1EA3

File: WooCommerce/src/main/kotlin/com/woocommerce/android/ui/blaze/creation/destination/BlazeCampaignCreationAdDestinationViewModel.kt [hunk 3]
```diff
@@ -47,21 +48,49 @@ class BlazeCampaignCreationAdDestinationViewModel @Inject constructor(
...
             isUrlDialogVisible = false
         )
     }
 
+    private fun getParameters(url: String): String {
+        return url.split("?")
+            .getOrNull(1)
+            ?.replace("&", "\n")
+            ?: return resourceProvider.getString(R.string.blaze_campaign_edit_ad_destination_empty_parameters_message)
+    }
```
Comment (JorgeMucientes) [inline WooCommerce/src/main/kotlin/com/woocommerce/android/ui/blaze/creation/destination/BlazeCampaignCreationAdDestinationViewModel.kt]: Minor suggestion, you can maybe reuse the `parseParameters()` function to make it slightly "more readable"? 
```kotlin
    private fun getParameters(url: String): String {
        return url.parseParameters().entries.joinToString(separator = "\n")
            .ifBlank { resourceProvider.getString(R.string.blaze_campaign_edit_ad_destination_empty_parameters_message) }
    }
```

It's more of a personal taste thing, so feel free to ignore this.

**Review by JorgeMucientes — APPROVED** [review]

**Comment by JorgeMucientes** [comment]
Great work @0nko. Everything works as expected and code looks good! I just left a couple questions, but aside from that this si ready to be merged. :shipit:

File: WooCommerce/src/main/kotlin/com/woocommerce/android/ui/blaze/creation/destination/BlazeCampaignCreationAdDestinationParametersScreen.kt (hunk 1 above)
Comment (0nko) [reply WooCommerce/src/main/kotlin/com/woocommerce/android/ui/blaze/creation/destination/BlazeCampaignCreationAdDestinationParametersScreen.kt:151]: Because of the animation. When I put it outside, the lazy column items are animated but the footer jumps.

File: WooCommerce/src/main/kotlin/com/woocommerce/android/ui/blaze/creation/destination/BlazeCampaignCreationAdDestinationParametersViewModel.kt (hunk 2 above)
Comment (0nko) [reply WooCommerce/src/main/kotlin/com/woocommerce/android/ui/blaze/creation/destination/BlazeCampaignCreationAdDestinationParametersViewModel.kt:21]: Yes, I've updated the behavior to match the web/iOS.

File: WooCommerce/src/main/kotlin/com/woocommerce/android/ui/blaze/creation/destination/BlazeCampaignCreationAdDestinationViewModel.kt (hunk 3 above)
Comment (0nko) [reply WooCommerce/src/main/kotlin/com/woocommerce/android/ui/blaze/creation/destination/BlazeCampaignCreationAdDestinationViewModel.kt]: Thanks for the suggestion, I updated the code.

**Comment by 0nko** [comment]
@JorgeMucientes Thanks for the review. I've addressed your feedback.
